### PR TITLE
Add script to get usernames of course participants

### LINKED ISSUE #20: Lagre lister over alle studenter i kurs:
kommandoen `lsng`kan brukes til å finne alle studentene som tar et kurs på ifi:
```
[olehelg@vetur ~]$ lsng sinf1000
bruker-drift
brukerto
brukertre
brukerfyr
```

En liten oppgave er å kjøre denne kommandoen for alle kurst i `course_list.json` (generert av `download-course-list.py`) og lagre resultatet i txt filer:
```
[olehelg@vetur ~]$ lsng sinf1000 > INF1000.txt
```

**OBS:** legg merke til at gruppene har små bokstaver, men kurskodene er med store bokstaver. I kurs hvor det er en bindestrek i kurskoden fjernes denne, så: `INF-BIO5121` blir 'sinfbio5121'

### PR DESCRIPTION
**Pull request by evestera**
Closes #20 

Needed for #21

### CODE REVIEW

**Comment by evestera** [comment]
Need to add some documentation and a Makefile entry before merging.

File: src/get_usernames.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import argparse
+import sys
+import os
+from file_funcs import load_json
+
+def get_args():
+    argparser = argparse.ArgumentParser(description='Get usernames of students taking courses this semester')
+    argparser.add_argument('--username', '-u', help='Username for login', type=str)
+    argparser.add_argument('--password', '-p', help='Password for login', type=str)
+    argparser.add_argument('--semester', '-s', help='Semester to read data for', type=str)
```
Comment (olehermanse) [inline src/get_usernames.py]: Misleading. You can only get usernames for current semester, so this option only decides where to read/write in local data.

> Semester to save data to

or similar.

**Comment by olehermanse** [comment]
@evestera Ready for review?

**Comment by evestera** [comment]
@olehermanse Sure. Should be good to go.

**Review by olehermanse — COMMENTED** [review]
Nice. One small request: add an argument for `--input-file`.

File: src/get_usernames.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import os
+from datetime import date, timedelta
+from file_funcs import load_json
+
+def get_args():
+    argparser = argparse.ArgumentParser(description='Get usernames of students taking courses this semester')
+    argparser.add_argument('--username', '-u', help='Username for login', type=str)
+    argparser.add_argument('--password', '-p', help='Password for login', type=str)
+    argparser.add_argument('--prev', help='Read data for the previous semester', action='store_true')
+    args = argparser.parse_args()
```
Comment (olehermanse) [inline src/get_usernames.py:23]: Would be useful to have an optional argument `--filename` or `--input` if you want to specify a different course list than `spring.json` or `fall.json`

**Review by olehermanse — APPROVED** [review]
Tested that this works. Well written. Nice that you included some tests. Merging now, can add the requested feature later.

**Pull request merged**